src/logic/exceptions/CodedException.ts

```diff
@@ -0,0 +1,37 @@
+import * as Sentry from '@sentry/react'
+import registry from './registry'
+import { IS_PRODUCTION } from 'src/utils/constants'
+
+export class CodedException extends Error {
+  public code: number
+  public uiMessage: string | undefined
```

Comment:
can we make them private and expose them through a getter?

Reply:
Why?

Reply:
once instantiated the message you could modify the code and iMessage, `instance.code = newCode`, is that what we want?

Reply:
That's a good point. I will add [`readonly`](https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier) to avoid that.